test/metabase/lib/remove_replace_test.cljc

```diff
@@ -734,3 +737,18 @@
         (testing "by name"
           (is (=? result
                   (lib/remove-join query' 0 (second (lib/joins query' 0))))))))))
+
+(deftest ^:parallel replace-join-test
+  (testing "#32026"
+    (let [query           (lib.tu/query-with-join)
+          [original-join] (lib/joins query)]
+      (is (=? {:lib/type :mbql/join, :alias "Cat", :fields :all, :lib/options {:lib/uuid string?}}
+              original-join))
+      (is (string? (lib.options/uuid original-join)))
+      (is (=? {:stages [{:joins [{:lib/type :mbql/join, :alias "Cat", :fields :all, :lib/options {:lib/uuid string?}}]}]}
+              query))
+      (let [new-join (lib/with-join-fields original-join :none)]
+        (is (=? {:lib/type :mbql/join, :alias "Cat", :fields :none, :lib/options {:lib/uuid string?}}
+                new-join))
+        (is (=? {:stages [{:joins [{:lib/type :mbql/join, :alias "Cat", :fields :none, :lib/options {:lib/uuid string?}}]}]}
+                (lib/replace-clause query original-join new-join)))))))
```

Comment:
There should be tests checking that invalidated parts get removed.

Reply:
Removed from this PR